src/timedelta_isoformat/__init__.py

```diff
@@ -178,7 +196,7 @@ def fromisoformat(cls, duration: str) -> "timedelta":
         :raises: `ValueError` with an explanatory message when parsing fails
         """
         try:
-            return cls(**dict(cls._from_duration(duration)))
+            return cls(**dict(c.astuple() for c in cls._from_duration(duration) if c.quantity))
```

Comment:
The only reason for the generator comprehension and `c.quantity` check here is that we want to allow parsing inputs that contain zero months-and-years ([`P0Y`](https://github.com/joesarre/timedelta-isoformat/blob/6ea64713110e50e59c259c1895948b2400151599/tests/test_timedelta_isoformat.py#L9), for example), even though `datetime.timedelta` itself cannot represent year-or-month fields.

We construct the arguments to build the `timedelta` from an iterator of tuple-pairs - a bit like `dict([("days", 1), ("hours", 2)])`.

If there were a way for [`astuple`](https://github.com/jayaddison/timedelta-isoformat/blob/dcdf8a855b414e6a18b20928acb60dd96740d034/src/timedelta_isoformat/__init__.py#L46-L47) to return an 'ignored' dict-field-tuple-pair (a tuple that would not result in creation of a key and value in the resulting dictionary), then we could emit that when the component's quantity is zero.

As far as I know, such a thing does not exist currently.

A weird workaround would be to return `weeks: 0` from `astuple` in those cases.  It's a bit sketchy and difficult-to-understand, but it does work.